### PR TITLE
Upgrade terraform provider cloudinit to v2.3.3

### LINKED ISSUE #121: [internal] Missing Automation: Upgrade checks
This provider has not been kept in sync with upstream upgrades.

https://github.com/hashicorp/terraform-provider-cloudinit/releases shows latest as last February. On our end it shows as [nearly two years](https://www.pulumi.com/registry/packages/cloudinit/)

Verify this provider is [correctly hooked up to upstream upgrade detection](https://github.com/pulumi/platform-providers-team/blob/main/playbooks/automating-provider-updates.md#upstream-provider-upgrade-automation). Enable if not; troubleshoot otherwise.

### LINKED ISSUE #187: Workflow failure: Check upstream upgrade
## Workflow Failure

[Check upstream upgrade](https://github.com/pulumi/pulumi-cloudinit/blob/master/.github/workflows/check-upstream-upgrade.yml) has failed. See the list of failures below:

- [2023-12-15T03:04:13.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7217420714)

- [2023-12-16T03:04:06.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7229385625)

- [2023-12-17T03:04:09.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7235955953)

- [2023-12-18T03:04:25.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7243070967)

- [2023-12-19T03:04:00.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7256813333)

- [2023-12-20T03:21:38.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7270445656)

- [2023-12-21T03:04:05.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7283470973)

- [2023-12-22T03:04:17.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7295715841)

- [2023-12-23T03:04:16.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7305810243)

- [2023-12-24T03:04:09.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7312136394)

- [2023-12-25T03:04:11.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7318632717)

- [2023-12-26T03:04:05.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7326151977)

- [2023-12-27T03:04:16.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7334628505)

- [2023-12-28T03:04:01.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7343974526)

- [2023-12-29T03:04:04.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7353307963)

- [2023-12-30T03:03:55.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7361674414)

- [2023-12-31T03:04:00.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7368013885)

- [2024-01-01T03:04:20.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7374483740)

- [2024-01-02T03:04:12.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7381806537)

- [2024-01-03T03:04:15.000Z](https://github.com/pulumi/pulumi-cloudinit/actions/runs/7392999498)

### PR DESCRIPTION
**Pull request by guineveresaenger**
Brings this provider out of the Dark Age.

Fixes #121. 
Fixes #187. 

- Move provider to plugin framework. Move to bridge token mapping via metadata.json. Update upstream ref as pseudoversion.
- make build sdks

### CODE REVIEW

File: provider/cmd/pulumi-resource-cloudinit/schema.json [hunk 1]
```diff
@@ -85,161 +93,173 @@
...
             },
-            "requiredInputs": [
-                "parts"
-            ],
             "stateInputs": {
                 "description": "Input properties used for looking up and filtering Config resources.\n",
                 "properties": {
                     "base64Encode": {
                         "type": "boolean",
-                        "willReplaceOnChanges": true
```
Comment (VenelinMartinov) [inline provider/cmd/pulumi-resource-cloudinit/schema.json:164]: This has been removed from quite a few resources, is this a concern?

File: provider/cmd/pulumi-resource-cloudinit/main.go [hunk 2]
```diff
@@ -17,17 +17,20 @@
 package main
 
 import (
+	"context"
 	_ "embed"
-
 	cloudinit "github.com/pulumi/pulumi-cloudinit/provider"
-	"github.com/pulumi/pulumi-cloudinit/provider/pkg/version"
-	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
+	//"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
```
Comment (VenelinMartinov) [inline provider/cmd/pulumi-resource-cloudinit/main.go:23]: commented out code

File: provider/resources.go [hunk 3]
```diff
@@ -63,28 +65,29 @@ func makeDataSource(mod string, res string) tokens.ModuleMember {
...
-		Resources: map[string]*tfbridge.ResourceInfo{
-			"cloudinit_config": {
-				Tok: makeResource(mainMod, "Config"),
-				Docs: &tfbridge.DocInfo{
-					Markdown: []byte(" "),
-				},
-				DeprecationMessage: "This resource is deprecated.\nPlease use the getConfig data source instead.",
-			},
-		},
+		//Resources: map[string]*tfbridge.ResourceInfo{
```
Comment (VenelinMartinov) [inline provider/resources.go:82]: commented out code

**Review by VenelinMartinov — APPROVED** [review]

File: provider/resources.go (hunk 3 above)
Comment (iwahbe) [reply provider/resources.go:82]: I think we can delete `Resources` and `DataSources` here.

File: provider/cmd/pulumi-resource-cloudinit/main.go [hunk 4]
```diff
@@ -17,17 +17,20 @@
...
+	"github.com/pulumi/pulumi-terraform-bridge/pf/tfbridge"
```
Comment (iwahbe) [inline provider/cmd/pulumi-resource-cloudinit/main.go:24]: If you made this change to `pulumi-resource-cloudinit/main.go`, you need to make an equivalent change to https://github.com/pulumi/pulumi-cloudinit/blob/5feceebee8c4cc719add05382d22bcfd1323f284/provider/cmd/pulumi-tfgen-cloudinit/main.go#L18.

**Review by iwahbe — COMMENTED** [review]
What happened here? Did we fail to hook this provider up to Zapier and then just forget about it?

I assume that we caught during the transition to our new upgrade check process (thanks @guineveresaenger!).

File: provider/cmd/pulumi-resource-cloudinit/schema.json (hunk 1 above)
Comment (guineveresaenger) [reply provider/cmd/pulumi-resource-cloudinit/schema.json:164]: I don't think so, but I don't have a lot of context here. This is also something users can control with our own `replaceOnChanges` option if it becomes an issue.

**Pull request merged**